operations/observability/mixins/IDE/rules/components/openvsx-proxy/alerts.libsonnet

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright (c) 2021 Gitpod GmbH. All rights reserved.
+ * Licensed under the MIT License. See License-MIT.txt in the project root for license information.
+ */
+
+{
+  prometheusAlerts+:: {
+    groups+: [
+      {
+        name: 'gitpod-component-openvsx-proxy-alerts',
+        rules: [
+          {
+            alert: 'GitpodOpenVSXRegistryDown',
+            labels: {
+              severity: 'critical',
+            },
+            'for': '20m',
+            annotations: {
+              runbook_url: 'https://github.com/gitpod-io/runbooks/blob/main/runbooks/GitpodOpenVsxRegistryDown.md',
+              summary: 'Open-VSX registry is possibly down',
+              description: 'Open-VSX registry is possibly down. We cannot pull VSCode extensions we don\'t have in our caches',
+            },
+            expr:
+              |||
+                sum(rate(gitpod_openvsx_proxy_requests_total{status=~"5\\d\\d|error"}[5m])) > 0.5
```

Comment:
we need to alert on error ration not rate i.e. something like
```
sum(
  rate(
    gitpod_openvsx_proxy_requests_total{status=~"5\\d\\d|error"}[5m]
  )
) /
sum(
  rate(
    gitpod_openvsx_proxy_requests_total[5m]
  )
)
```

Error rate is misleading, i.e. seen it spikes with growth of throughput. Ratio stays stable.

Why 0.5? It is quite critical, it should be around 99% successful, so if it is over 1% is already bad, i.e `> 0.01`

Reply:
You can read up about alerting on SLIs in https://www.oreilly.com/library/view/sre-with-java/9781492073918/ch04.html Look for `Service Level Indicators for Every Java Microservice`. The book about Java, but it is not really about Java 😆